src/coreclr/nativeaot/System.Private.Reflection.Core/src/System/Reflection/Runtime/Assemblies/NativeFormat/NativeFormatRuntimeAssembly.GetTypeCore.CaseInsensitive.cs

```diff
@@ -99,8 +99,7 @@ internal sealed override RuntimeTypeInfo GetTypeCoreCaseInsensitive(string fullN
                     foreach (TypeDefinitionHandle typeDefinitionHandle in namespaceDefinition.TypeDefinitions)
                     {
                         string fullName = ns + typeDefinitionHandle.GetTypeDefinition(reader).Name.GetString(reader).ToLowerInvariant();
-                        QHandle existingValue;
-                        if (!dict.TryGetValue(fullName, out existingValue))
+                        if (!dict.TryGetValue(fullName, out _))
```

Comment:
Regular Dictionary depends on complex generic constructs that call back into type loader. Using regular Dictionary in the NativeAOT type loader tends to cause infinite recursion. Some uses of the LowLevelDictionary can be probably replaced with Dictionary, but I do not think it would be possible to replace all of them.

The problem with larger code size is there as well.